pkg/watcher/watcher.go

```diff
@@ -117,34 +119,50 @@ func (w *Watcher) StartWatching() {
 	}
 	w.mutex.RUnlock()
 
-	windowWatcher := func(duration string) {
-		for {
-			curWindow, metric := w.getCurrentWindow(duration)
-			fetchedMetricsMap, err := w.client.FetchAllHostsMetrics(curWindow)
-			if err != nil {
-				log.Errorf("received error while fetching metrics: %v ", err)
-				continue
-			}
-			log.Debugf("fetched metrics for window: %v", curWindow)
-			metricsMap := make(map[string]NodeMetrics)
-			for host, fetchedMetrics := range fetchedMetricsMap {
-				nodeMetric := NodeMetrics{
-					Metrics: make([]Metric, len(fetchedMetrics)),
-				}
-				copy(nodeMetric.Metrics, fetchedMetrics)
-				metricsMap[host] = nodeMetric
-			}
-			watcherMetrics := &WatcherMetrics{Timestamp: time.Now().Unix(),
-				Data: Data{NodeMetricsMap: metricsMap},
+	fetchOnce := func(duration string, datasource string) {
```

Comment:
Instead of defining source here and reflecting on client struct name below, can we instead define a new method in `FetcherClient` interface called `Name` which returns the client/source name. Then all you need to do in `Source` initialization on line `141` below is call `w.client.Name()`. Also, you wouldn't need a switch/case block then.

Reply:
This will also take care of your TODO comment elsewhere of defining each client's name in respective file as a const.

Reply:
Ok, I agree. Let's do it this way.

Reply:
Besides, we miss a field to indicate if the metric is `average` or `std`. I am now using the `name` field for that. Do we consistently define that for each metric?

Reply:
Oh good point, feel free to add something like `operator`/`aggregation` etc. in JSON schema for this. `name` field is name of series/table in the provider db and serves like a "primary key" for us to relate.